tokio/src/net/windows/named_pipe.rs

```diff
@@ -998,15 +998,18 @@ impl NamedPipeClientOptions {
 
     /// Open the named pipe identified by `addr`.
     ///
-    /// This is the same as [`create`][NamedPipeClientOptions::create] except that
-    /// it supports providing security attributes.
+    /// This is the same as [`create`] except that it supports providing
+    /// security attributes.
     ///
     /// # Safety
     ///
-    /// The caller must ensure that `attrs` points to an initialized instance
-    /// of a [`SECURITY_ATTRIBUTES`] structure.
+    /// The caller must ensure that `attrs` points to an initialized instance of
+    /// a [`SECURITY_ATTRIBUTES`] structure. This argument *can* be set to null
+    /// and be ignored. But the caller should then instead prefer to use
+    /// [`create`].
```

Comment:
I think it's better to just say what the behavior is. It could be reasonable to pass a null pointer here, e.g. if the actual value is determined at runtime.
```suggestion
    /// The `attrs` argument must either be null or point at a valid instance
    /// of the [`SECURITY_ATTRIBUTES`] structure. If the argument is null,
    /// the behavior is identical to calling the [`create`] method.
```